Jetchat/app/src/main/java/com/example/compose/jetchat/PalmWrapper.kt

```diff
@@ -0,0 +1,135 @@
+package com.example.compose.jetchat
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.ai.generativelanguage.v1beta2.DiscussServiceClient
+import com.google.ai.generativelanguage.v1beta2.DiscussServiceSettings
+import com.google.ai.generativelanguage.v1beta2.Example
+import com.google.ai.generativelanguage.v1beta2.GenerateMessageRequest
+import com.google.ai.generativelanguage.v1beta2.Message
+import com.google.ai.generativelanguage.v1beta2.MessagePrompt
+import com.google.api.gax.core.FixedCredentialsProvider
+import com.google.api.gax.grpc.InstantiatingGrpcChannelProvider
+import com.google.api.gax.rpc.FixedHeaderProvider
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class PalmWrapper {
+    /** Matches the OpenAI implementation */
+    suspend fun chat(message: String): String {
+        return sendMessage (message)
+    }
+
+    private val _messages = MutableStateFlow<List<Message>>(value = listOf())
+    val messages: StateFlow<List<Message>>
+        get() = _messages
```

Comment:
I was assuming you made a `StateFlow` because there were some issues with the messages updating in the UI, but if not then I would even suggest just removing the backing field and having `val message = mutableListOf<Message>()` - you may not even need a `MutableStateList`
```suggestion
    private val _messages = mutableStateList<Message>()
    val messages: SnapshotStateList<Message>
        get() = _messages
```